File: I1_MVPAndRetrofit/app/src/main/java/com/example/tsvetelinastoyanova/weatherreportrevisited/data/source/remote/RetrofitClientInstance.java
package com.example.tsvetelinastoyanova.weatherreportrevisited.data.source.remote;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClientInstance {
    private static Retrofit retrofit;
    private static final String BASE_URL_CITY = "http://api.openweathermap.org/data/2.5/";

    public static Retrofit getRetrofitInstance() {
        if (retrofit == null) {
            retrofit = new retrofit2.Retrofit.Builder()
                    .baseUrl(BASE_URL_CITY)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }

        return retrofit;
    }
}

File: I1_MVPAndRetrofit_Kotlin/app/src/main/java/com/example/tsvetelinastoyanova/weatherapp/data/source/remote/RetrofitClientInstance.kt
package com.example.tsvetelinastoyanova.weatherapp.data.source.remote

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory


object RetrofitClientInstance {
    const val BASE_URL_CITY = "http://api.openweathermap.org/data/2.5/"
    val instance: Retrofit = retrofit2.Retrofit.Builder()
            .baseUrl(BASE_URL_CITY)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
}